File: leetcode/tree/symmetric_tree.py
def isSymmetric(self, root: Optional[TreeNode]) -> bool:
    return self.f(root.left, root.right)

def f(self, l, r):
    if not l or not r:
        return l == r
    else:
        return l.val == r.val and self.f(l.right, r.left) and self.f(l.left, r.right)

File: leetcode/tree/symmetric_tree.cpp
bool isSymmetric(TreeNode* root) {
	if (f(root->left, root->right))
		return true;
	else
		return false;
}
bool f(TreeNode* l, TreeNode* r) {
	if (!r || !l) 
		return l == r;
	else {
		return r->val == l->val && f(l->left, r->right) && f(l->right, r->left);
	}
}
